src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -2946,34 +2939,46 @@ bool gfx_movemem_handler_otr(F3DGfx** cmd0) {
     return false;
 }
 
-int16_t gfx_create_shader(const std::string& path) {
-    std::shared_ptr<Ship::ResourceInitData> initData = std::make_shared<Ship::ResourceInitData>();
-    initData->Path = path;
-    initData->IsCustom = false;
-    initData->ByteOrder = Ship::Endianness::Native;
-    auto shader = Ship::Context::GetInstance()->GetResourceManager()->GetArchiveManager()->LoadFile(path);
-    if (shader == nullptr || !shader->IsLoaded) {
-        return -1;
+int16_t gfx_search_existing_shader(const char* vertex, const char* fragment) {
+    ShaderMod shader = { vertex, fragment };
+    for (int i = 0; i < shader_ids.size(); i++) {
```

Comment:
I am weary of a linear search. This array will grow in size, especially for longer play sessions. Consider using a map. If we have a LRU cache in LUS, this may be a good place for it.

Reply:
This is an rather old comment, so I'm not too sure of the context now. I can easily see custom shaders getting to the several hundred mark, especially as permutations accumulate. It's impossible to fully know without usage data. The "more performant option" would be the LRU cache I mentioned, since I assume the set of shaders used on a frame-to-frame basis is largely unchanged.